In [2]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.


In [3]:
import pandas as pd

### Add CSV (comma-separated values) to pandas.

In [4]:
### Read CSV File

data =pd.read_csv('addresses.csv')

In [5]:
data

id  location_id                   address_1  address_2  \
0    1            1       2600 Middlefield Road        NaN   
1    2            2            24 Second Avenue        NaN   
2    3            3            24 Second Avenue        NaN   
3    4            4            24 Second Avenue        NaN   
4    5            5            24 Second Avenue        NaN   
5    6            6           800 Middle Avenue        NaN   
6    7            7              500 Arbor Road        NaN   
7    8            8           800 Middle Avenue        NaN   
8    9            9       2510 Middlefield Road        NaN   
9   10           10       1044 Middlefield Road        NaN   
10  11           11         2140 Euclid Avenue.        NaN   
11  12           12       1044 Middlefield Road  2nd Floor   
12  13           13         399 Marine Parkway.        NaN   
13  14           14          660 Veterans Blvd.        NaN   
14  15           15        1500 Valencia Street        NaN   
15  16           16         1161 South Bernardo        NaN   
16  17           17     409 South Spruce Avenue        NaN   
17  18           18            114 Fifth Avenue        NaN   
18  19           19         19 West 39th Avenue        NaN   
19  20           21          123 El Camino Real        NaN   
20  21           22  2013 Avenue of the fellows  Suite 100   

                   city state_province postal_code country  
0          Redwood City             CA       94063      US  
1             San Mateo             CA       94401      US  
2             San Mateo             CA       94403      US  
3             San Mateo             CA       94401      US  
4             San Mateo             CA       94401      US  
5            Menlo Park             CA  94025-9881      US  
6            Menlo Park             CA       94025      US  
7            Menlo Park             CA  94025-9881      US  
8          Redwood City             CA       94063      US  
9          Redwood City             CA       94063      US  
10         Redwood City             CA       94061      US  
11         Redwood City             CA       94063      US  
12         Redwood City             CA       94065      US  
13         Redwood City             CA       94063      US  
14        San Francisco             CA       94110      US  
15            Sunnyvale             CA       94087      US  
16  South San Francisco             CA       94080      US  
17         Redwood City             CA       94063      US  
18            San Mateo             CA       94403      US  
19              Belmont             CA       94002      US  
20        San Francisco             CA       94103      US

In [6]:
type(data)

pandas.core.frame.DataFrame

### What is a DataFrame in pandas?

What is a DataFrame? A Pandas DataFrame is a 2 dimensional data structure, like a 2 dimensional array, or a table with rows and columns.

### DataFrame vs series in pandas?

Series can only contain single list with index, whereas dataframe can be made of more than one series or we can say that a dataframe is a collection of series that can be used to analyse the data.

### DataFrame - head() function

The head() function is used to get the first n rows. This function returns the first n rows for the object based on position. It is useful for quickly testing if your object has the right type of data in it. Number of rows to select.

Type/Default Value = int ==== Default Value: 5

Syntax: DataFrame.head(self, n=5)

https://www.w3resource.com/pandas/dataframe/dataframe-head.php#:~:text=DataFrame%20%2D%20head()%20function,type%20of%20data%20in%20it.&text=Number%20of%20rows%20to%20select.

In [7]:
data.head()

id  location_id              address_1 address_2          city  \
0   1            1  2600 Middlefield Road       NaN  Redwood City   
1   2            2       24 Second Avenue       NaN     San Mateo   
2   3            3       24 Second Avenue       NaN     San Mateo   
3   4            4       24 Second Avenue       NaN     San Mateo   
4   5            5       24 Second Avenue       NaN     San Mateo   

  state_province postal_code country  
0             CA       94063      US  
1             CA       94401      US  
2             CA       94403      US  
3             CA       94401      US  
4             CA       94401      US

### DataFrame - tail() function :

The tail() function is used to get the last n rows.
Syntax: DataFrame.tail(self, n=5)

In [8]:
data.tail()

id  location_id                   address_1  address_2  \
16  17           17     409 South Spruce Avenue        NaN   
17  18           18            114 Fifth Avenue        NaN   
18  19           19         19 West 39th Avenue        NaN   
19  20           21          123 El Camino Real        NaN   
20  21           22  2013 Avenue of the fellows  Suite 100   

                   city state_province postal_code country  
16  South San Francisco             CA       94080      US  
17         Redwood City             CA       94063      US  
18            San Mateo             CA       94403      US  
19              Belmont             CA       94002      US  
20        San Francisco             CA       94103      US

In [9]:
data.tail(7)

id  location_id                   address_1  address_2  \
14  15           15        1500 Valencia Street        NaN   
15  16           16         1161 South Bernardo        NaN   
16  17           17     409 South Spruce Avenue        NaN   
17  18           18            114 Fifth Avenue        NaN   
18  19           19         19 West 39th Avenue        NaN   
19  20           21          123 El Camino Real        NaN   
20  21           22  2013 Avenue of the fellows  Suite 100   

                   city state_province postal_code country  
14        San Francisco             CA       94110      US  
15            Sunnyvale             CA       94087      US  
16  South San Francisco             CA       94080      US  
17         Redwood City             CA       94063      US  
18            San Mateo             CA       94403      US  
19              Belmont             CA       94002      US  
20        San Francisco             CA       94103      US

In [10]:
data.columns

Index(['id', 'location_id', 'address_1', 'address_2', 'city', 'state_province',
       'postal_code', 'country'],
      dtype='object')

In [11]:
list(data.columns)

['id',
 'location_id',
 'address_1',
 'address_2',
 'city',
 'state_province',
 'postal_code',
 'country']

In [12]:
data.head(3)

id  location_id              address_1 address_2          city  \
0   1            1  2600 Middlefield Road       NaN  Redwood City   
1   2            2       24 Second Avenue       NaN     San Mateo   
2   3            3       24 Second Avenue       NaN     San Mateo   

  state_province postal_code country  
0             CA       94063      US  
1             CA       94401      US  
2             CA       94403      US

In [13]:
### Looking for  'id' , 'address_1' :

data [['id' , 'address_1']]

id                   address_1
0    1       2600 Middlefield Road
1    2            24 Second Avenue
2    3            24 Second Avenue
3    4            24 Second Avenue
4    5            24 Second Avenue
5    6           800 Middle Avenue
6    7              500 Arbor Road
7    8           800 Middle Avenue
8    9       2510 Middlefield Road
9   10       1044 Middlefield Road
10  11         2140 Euclid Avenue.
11  12       1044 Middlefield Road
12  13         399 Marine Parkway.
13  14          660 Veterans Blvd.
14  15        1500 Valencia Street
15  16         1161 South Bernardo
16  17     409 South Spruce Avenue
17  18            114 Fifth Avenue
18  19         19 West 39th Avenue
19  20          123 El Camino Real
20  21  2013 Avenue of the fellows

In [14]:
data[['postal_code' , 'country']]

postal_code country
0        94063      US
1        94401      US
2        94403      US
3        94401      US
4        94401      US
5   94025-9881      US
6        94025      US
7   94025-9881      US
8        94063      US
9        94063      US
10       94061      US
11       94063      US
12       94065      US
13       94063      US
14       94110      US
15       94087      US
16       94080      US
17       94063      US
18       94403      US
19       94002      US
20       94103      US

In [15]:
### pandas.core.series.Series

data ['state_province']

0     CA
1     CA
2     CA
3     CA
4     CA
5     CA
6     CA
7     CA
8     CA
9     CA
10    CA
11    CA
12    CA
13    CA
14    CA
15    CA
16    CA
17    CA
18    CA
19    CA
20    CA
Name: state_province, dtype: object

In [16]:
### pandas.core.frame.DataFrame

data [['state_province']]

state_province
0              CA
1              CA
2              CA
3              CA
4              CA
5              CA
6              CA
7              CA
8              CA
9              CA
10             CA
11             CA
12             CA
13             CA
14             CA
15             CA
16             CA
17             CA
18             CA
19             CA
20             CA

In [17]:
type(data ['state_province'])

pandas.core.series.Series

In [18]:
type(data [['state_province']])

pandas.core.frame.DataFrame

In [19]:
data[['city' , 'location_id']]

city  location_id
0          Redwood City            1
1             San Mateo            2
2             San Mateo            3
3             San Mateo            4
4             San Mateo            5
5            Menlo Park            6
6            Menlo Park            7
7            Menlo Park            8
8          Redwood City            9
9          Redwood City           10
10         Redwood City           11
11         Redwood City           12
12         Redwood City           13
13         Redwood City           14
14        San Francisco           15
15            Sunnyvale           16
16  South San Francisco           17
17         Redwood City           18
18            San Mateo           19
19              Belmont           21
20        San Francisco           22

### Add Excel file to python :

- xlrd : Library for developers to extract data from Microsoft Excel (tm) .xls spreadsheet files.
- https://anaconda.org/anaconda/xlrd
- https://bobbyhadz.com/blog/python-no-module-named-xlrd

In [20]:
pip install xlrd

Note: you may need to restart the kernel to use updated packages.


In [21]:
pd.read_excel('airline.xls')

YEAR       Y      W       R      L      K
0   1948   1.214  0.243  0.1454  1.415  0.612
1   1949   1.354  0.260  0.2181  1.384  0.559
2   1950   1.569  0.278  0.3157  1.388  0.573
3   1951   1.948  0.297  0.3940  1.550  0.564
4   1952   2.265  0.310  0.3559  1.802  0.574
5   1953   2.731  0.322  0.3593  1.926  0.711
6   1954   3.025  0.335  0.4025  1.964  0.776
7   1955   3.562  0.350  0.3961  2.116  0.827
8   1956   3.979  0.361  0.3822  2.435  0.800
9   1957   4.420  0.379  0.3045  2.707  0.921
10  1958   4.563  0.391  0.3284  2.706  1.067
11  1959   5.385  0.426  0.3856  2.846  1.083
12  1960   5.554  0.441  0.3193  3.089  1.481
13  1961   5.465  0.460  0.3079  3.122  1.736
14  1962   5.825  0.485  0.3783  3.184  1.926
15  1963   6.876  0.506  0.4180  3.263  2.041
16  1964   7.823  0.538  0.5163  3.412  1.997
17  1965   9.120  0.564  0.5879  3.623  2.257
18  1966  10.512  0.586  0.5369  4.074  2.742
19  1967  13.020  0.622  0.4443  4.710  3.564
20  1968  15.261  0.666  0.3052  5.217  4.767
21  1969  16.313  0.731  0.2332  5.569  6.511
22  1970  16.002  0.831  0.1883  5.495  7.627
23  1971  15.876  0.906  0.2023  5.334  8.673
24  1972  16.662  1.000  0.2506  5.345  8.331
25  1973  17.014  1.056  0.2668  5.662  8.557
26  1974  19.305  1.131  0.2664  5.729  9.508
27  1975  18.721  1.247  0.2301  5.722  9.062
28  1976  19.250  1.375  0.3452  5.762  8.262
29  1977  20.647  1.544  0.4508  5.877  7.474
30  1978  22.726  1.703  0.5877  6.108  7.104
31  1979  23.619  1.779  0.5346  6.852  6.874

### Open sheet2 or sheet3 in Excel file insted of sheet1 :

https://developers.refinitiv.com/en/article-catalog/article/how-to-enable-autocompletion-in-jupyter-notebook-application

In [22]:
pd.read_excel('airline.xls' , sheet_name = "Sheet2")

1956.0000  3.9790     0.3610     0.3822     2.4350     0.8000   
0        1957      4.420      0.379     0.3045      2.707     0.9210
1        1958      4.563      0.391     0.3284      2.706     1.0670
2        1959      5.385      0.426     0.3856      2.846     1.0830
3        1960      5.554      0.441     0.3193      3.089     1.4810
4        1961      5.465      0.460     0.3079      3.122     1.7360
5        1962      5.825      0.485     0.3783      3.184     1.9260
6        1963      6.876      0.506     0.4180      3.263     2.0410
7        1964      7.823      0.538     0.5163      3.412     1.9970
8        1965      9.120      0.564     0.5879      3.623     2.2570
9        1966     10.512      0.586     0.5369      4.074     2.7420
10       1967     13.020      0.622     0.4443      4.710     3.5640
11       1968     15.261      0.666     0.3052      5.217     4.7670
12       1969     16.313      0.731     0.2332      5.569     6.5110
13       1970     16.002      0.831     0.1883      5.495     7.6270
14       1971     15.876      0.906     0.2023      5.334     8.6730
15       1972     16.662      1.000     0.2506      5.345     8.3310
16       1973     17.014      1.056     0.2668      5.662     9.9999

### Hold shift + Tab and see Signature 

input sheet_name for see sheet2 on excel sheet

In [23]:
pd.read_excel("airline.xls" , sheet_name = "Sheet2")

1956.0000  3.9790     0.3610     0.3822     2.4350     0.8000   
0        1957      4.420      0.379     0.3045      2.707     0.9210
1        1958      4.563      0.391     0.3284      2.706     1.0670
2        1959      5.385      0.426     0.3856      2.846     1.0830
3        1960      5.554      0.441     0.3193      3.089     1.4810
4        1961      5.465      0.460     0.3079      3.122     1.7360
5        1962      5.825      0.485     0.3783      3.184     1.9260
6        1963      6.876      0.506     0.4180      3.263     2.0410
7        1964      7.823      0.538     0.5163      3.412     1.9970
8        1965      9.120      0.564     0.5879      3.623     2.2570
9        1966     10.512      0.586     0.5369      4.074     2.7420
10       1967     13.020      0.622     0.4443      4.710     3.5640
11       1968     15.261      0.666     0.3052      5.217     4.7670
12       1969     16.313      0.731     0.2332      5.569     6.5110
13       1970     16.002      0.831     0.1883      5.495     7.6270
14       1971     15.876      0.906     0.2023      5.334     8.6730
15       1972     16.662      1.000     0.2506      5.345     8.3310
16       1973     17.014      1.056     0.2668      5.662     9.9999

### How can we read head item in excell sheet ? ( 1956.0000	3.9790	0.3610	0.3822	2.4350	0.8000)

go to signature and add  (header: 'int | Sequence[int] | None' = 0) 


In [24]:
pd.read_excel("airline.xls" , sheet_name = "Sheet2" , header = None)

0       1      2       3      4       5
0   1956   3.979  0.361  0.3822  2.435  0.8000
1   1957   4.420  0.379  0.3045  2.707  0.9210
2   1958   4.563  0.391  0.3284  2.706  1.0670
3   1959   5.385  0.426  0.3856  2.846  1.0830
4   1960   5.554  0.441  0.3193  3.089  1.4810
5   1961   5.465  0.460  0.3079  3.122  1.7360
6   1962   5.825  0.485  0.3783  3.184  1.9260
7   1963   6.876  0.506  0.4180  3.263  2.0410
8   1964   7.823  0.538  0.5163  3.412  1.9970
9   1965   9.120  0.564  0.5879  3.623  2.2570
10  1966  10.512  0.586  0.5369  4.074  2.7420
11  1967  13.020  0.622  0.4443  4.710  3.5640
12  1968  15.261  0.666  0.3052  5.217  4.7670
13  1969  16.313  0.731  0.2332  5.569  6.5110
14  1970  16.002  0.831  0.1883  5.495  7.6270
15  1971  15.876  0.906  0.2023  5.334  8.6730
16  1972  16.662  1.000  0.2506  5.345  8.3310
17  1973  17.014  1.056  0.2668  5.662  9.9999

In [25]:
### Change Colume name to 'A' , 'B' , 'C' , 'D' ,'E' , 'F'

pd.read_excel("airline.xls" , sheet_name = "Sheet2" , header = None , names=['A' , 'B' , 'C' , 'D' ,'E' , 'F'])

A       B      C       D      E       F
0   1956   3.979  0.361  0.3822  2.435  0.8000
1   1957   4.420  0.379  0.3045  2.707  0.9210
2   1958   4.563  0.391  0.3284  2.706  1.0670
3   1959   5.385  0.426  0.3856  2.846  1.0830
4   1960   5.554  0.441  0.3193  3.089  1.4810
5   1961   5.465  0.460  0.3079  3.122  1.7360
6   1962   5.825  0.485  0.3783  3.184  1.9260
7   1963   6.876  0.506  0.4180  3.263  2.0410
8   1964   7.823  0.538  0.5163  3.412  1.9970
9   1965   9.120  0.564  0.5879  3.623  2.2570
10  1966  10.512  0.586  0.5369  4.074  2.7420
11  1967  13.020  0.622  0.4443  4.710  3.5640
12  1968  15.261  0.666  0.3052  5.217  4.7670
13  1969  16.313  0.731  0.2332  5.569  6.5110
14  1970  16.002  0.831  0.1883  5.495  7.6270
15  1971  15.876  0.906  0.2023  5.334  8.6730
16  1972  16.662  1.000  0.2506  5.345  8.3310
17  1973  17.014  1.056  0.2668  5.662  9.9999

In [26]:
pd.read_excel("airline.xls" , sheet_name = "Sheet2" , header = None , names=['A' , 'B' , 'C' , 'D' ])

A       B      C       D
1956 3.979   0.361  0.3822  2.435  0.8000
1957 4.420   0.379  0.3045  2.707  0.9210
1958 4.563   0.391  0.3284  2.706  1.0670
1959 5.385   0.426  0.3856  2.846  1.0830
1960 5.554   0.441  0.3193  3.089  1.4810
1961 5.465   0.460  0.3079  3.122  1.7360
1962 5.825   0.485  0.3783  3.184  1.9260
1963 6.876   0.506  0.4180  3.263  2.0410
1964 7.823   0.538  0.5163  3.412  1.9970
1965 9.120   0.564  0.5879  3.623  2.2570
1966 10.512  0.586  0.5369  4.074  2.7420
1967 13.020  0.622  0.4443  4.710  3.5640
1968 15.261  0.666  0.3052  5.217  4.7670
1969 16.313  0.731  0.2332  5.569  6.5110
1970 16.002  0.831  0.1883  5.495  7.6270
1971 15.876  0.906  0.2023  5.334  8.6730
1972 16.662  1.000  0.2506  5.345  8.3310
1973 17.014  1.056  0.2668  5.662  9.9999

In [27]:
### save file :

data1 =pd.read_excel("airline.xls" , sheet_name = "Sheet2" , header = None , names=['A' , 'B' , 'C' , 'D' ,'E' , 'F'])

In [28]:
data1

A       B      C       D      E       F
0   1956   3.979  0.361  0.3822  2.435  0.8000
1   1957   4.420  0.379  0.3045  2.707  0.9210
2   1958   4.563  0.391  0.3284  2.706  1.0670
3   1959   5.385  0.426  0.3856  2.846  1.0830
4   1960   5.554  0.441  0.3193  3.089  1.4810
5   1961   5.465  0.460  0.3079  3.122  1.7360
6   1962   5.825  0.485  0.3783  3.184  1.9260
7   1963   6.876  0.506  0.4180  3.263  2.0410
8   1964   7.823  0.538  0.5163  3.412  1.9970
9   1965   9.120  0.564  0.5879  3.623  2.2570
10  1966  10.512  0.586  0.5369  4.074  2.7420
11  1967  13.020  0.622  0.4443  4.710  3.5640
12  1968  15.261  0.666  0.3052  5.217  4.7670
13  1969  16.313  0.731  0.2332  5.569  6.5110
14  1970  16.002  0.831  0.1883  5.495  7.6270
15  1971  15.876  0.906  0.2023  5.334  8.6730
16  1972  16.662  1.000  0.2506  5.345  8.3310
17  1973  17.014  1.056  0.2668  5.662  9.9999

In [29]:
### Pandas to_csv() : https://www.digitalocean.com/community/tutorials/pandas-to_csv-convert-dataframe-to-csv

data1.to_csv("Pandas_Test")

In [30]:
### Signature: we want to change index as False insted of True (check Signature) 

data1.to_csv("Pandas_Test_1.csv" , index = False )

In [31]:
data1.to_excel("Pandas_Test_excel.xlsx" , header = False) 

## give me a first file and last file :

In [32]:
data1.head()

A      B      C       D      E      F
0  1956  3.979  0.361  0.3822  2.435  0.800
1  1957  4.420  0.379  0.3045  2.707  0.921
2  1958  4.563  0.391  0.3284  2.706  1.067
3  1959  5.385  0.426  0.3856  2.846  1.083
4  1960  5.554  0.441  0.3193  3.089  1.481

In [33]:
type(data1)

pandas.core.frame.DataFrame

In [34]:
data1.columns

Index(['A', 'B', 'C', 'D', 'E', 'F'], dtype='object')

In [35]:
list(data1.columns)

['A', 'B', 'C', 'D', 'E', 'F']

In [36]:
## fetch B and C :

data1[["B" , "C"]]

B      C
0    3.979  0.361
1    4.420  0.379
2    4.563  0.391
3    5.385  0.426
4    5.554  0.441
5    5.465  0.460
6    5.825  0.485
7    6.876  0.506
8    7.823  0.538
9    9.120  0.564
10  10.512  0.586
11  13.020  0.622
12  15.261  0.666
13  16.313  0.731
14  16.002  0.831
15  15.876  0.906
16  16.662  1.000
17  17.014  1.056

In [37]:
data.tail()

id  location_id                   address_1  address_2  \
16  17           17     409 South Spruce Avenue        NaN   
17  18           18            114 Fifth Avenue        NaN   
18  19           19         19 West 39th Avenue        NaN   
19  20           21          123 El Camino Real        NaN   
20  21           22  2013 Avenue of the fellows  Suite 100   

                   city state_province postal_code country  
16  South San Francisco             CA       94080      US  
17         Redwood City             CA       94063      US  
18            San Mateo             CA       94403      US  
19              Belmont             CA       94002      US  
20        San Francisco             CA       94103      US

In [38]:
## Titanic CSV file from HTML : 

pd.read_csv("https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv")

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [39]:
### Instal lxml in jupyter : 

import pandas as pd

!pip install lxml


In [40]:
 ### other website for load data : (read_html) 
    
pd.read_html("https://www.basketball-reference.com/leagues/NBA_2022_totals.html")

[      Rk             Player Pos Age   Tm   G  GS    MP   FG   FGA  ...   FT%  \
 0      1   Precious Achiuwa   C  22  TOR  73  28  1725  265   603  ...  .595   
 1      2       Steven Adams   C  28  MEM  76  75  1999  210   384  ...  .543   
 2      3        Bam Adebayo   C  24  MIA  56  56  1825  406   729  ...  .753   
 3      4       Santi Aldama  PF  21  MEM  32   0   360   53   132  ...  .625   
 4      5  LaMarcus Aldridge   C  36  BRK  47  12  1050  252   458  ...  .873   
 ..   ...                ...  ..  ..  ...  ..  ..   ...  ...   ...  ...   ...   
 837  601     Thaddeus Young  PF  33  TOR  26   0   475   67   144  ...  .481   
 838  602         Trae Young  PG  23  ATL  76  76  2652  711  1544  ...  .904   
 839  603     Omer Yurtseven   C  23  MIA  56  12   706  130   247  ...  .623   
 840  604        Cody Zeller   C  29  POR  27   0   355   51    90  ...  .776   
 841  605        Ivica Zubac   C  24  LAC  76  76  1852  310   495  ...  .727   
 
      ORB  DRB  TRB  AST S

In [41]:
data2 = pd.read_html("https://www.basketball-reference.com/leagues/NBA_2022_totals.html")

In [42]:
data2

[      Rk             Player Pos Age   Tm   G  GS    MP   FG   FGA  ...   FT%  \
 0      1   Precious Achiuwa   C  22  TOR  73  28  1725  265   603  ...  .595   
 1      2       Steven Adams   C  28  MEM  76  75  1999  210   384  ...  .543   
 2      3        Bam Adebayo   C  24  MIA  56  56  1825  406   729  ...  .753   
 3      4       Santi Aldama  PF  21  MEM  32   0   360   53   132  ...  .625   
 4      5  LaMarcus Aldridge   C  36  BRK  47  12  1050  252   458  ...  .873   
 ..   ...                ...  ..  ..  ...  ..  ..   ...  ...   ...  ...   ...   
 837  601     Thaddeus Young  PF  33  TOR  26   0   475   67   144  ...  .481   
 838  602         Trae Young  PG  23  ATL  76  76  2652  711  1544  ...  .904   
 839  603     Omer Yurtseven   C  23  MIA  56  12   706  130   247  ...  .623   
 840  604        Cody Zeller   C  29  POR  27   0   355   51    90  ...  .776   
 841  605        Ivica Zubac   C  24  LAC  76  76  1852  310   495  ...  .727   
 
      ORB  DRB  TRB  AST S

In [43]:
type(data2)

list

In [44]:
len(data2)

1

In [45]:
### same data set like " html adress " 

data2[0]

Rk             Player Pos Age   Tm   G  GS    MP   FG   FGA  ...   FT%  \
0      1   Precious Achiuwa   C  22  TOR  73  28  1725  265   603  ...  .595   
1      2       Steven Adams   C  28  MEM  76  75  1999  210   384  ...  .543   
2      3        Bam Adebayo   C  24  MIA  56  56  1825  406   729  ...  .753   
3      4       Santi Aldama  PF  21  MEM  32   0   360   53   132  ...  .625   
4      5  LaMarcus Aldridge   C  36  BRK  47  12  1050  252   458  ...  .873   
..   ...                ...  ..  ..  ...  ..  ..   ...  ...   ...  ...   ...   
837  601     Thaddeus Young  PF  33  TOR  26   0   475   67   144  ...  .481   
838  602         Trae Young  PG  23  ATL  76  76  2652  711  1544  ...  .904   
839  603     Omer Yurtseven   C  23  MIA  56  12   706  130   247  ...  .623   
840  604        Cody Zeller   C  29  POR  27   0   355   51    90  ...  .776   
841  605        Ivica Zubac   C  24  LAC  76  76  1852  310   495  ...  .727   

     ORB  DRB  TRB  AST STL BLK  TOV   PF   PTS  
0    146  327  473   82  37  41   84  151   664  
1    349  411  760  256  65  60  115  153   528  
2    137  427  564  190  80  44  148  171  1068  
3     33   54   87   21   6  10   16   36   132  
4     73  185  258   42  14  47   44   78   607  
..   ...  ...  ...  ...  ..  ..  ...  ...   ...  
837   40   75  115   45  31  11   22   43   164  
838   50  234  284  737  72   7  303  128  2155  
839   85  209  294   49  17  20   41   84   299  
840   50   75  125   22   8   6   19   56   140  
841  217  427  644  120  36  77  114  203   785  

[842 rows x 30 columns]

In [46]:
### save file :

data2[0].to_csv("2021-22 NBA Player Stats.csv")

In [47]:
pd.read_csv("Vehicle Specifications.csv")

normalized-losses         make fuel-type aspiration num-of-doors  \
0                  NaN  alfa-romero       gas        std          two   
1                  NaN  alfa-romero       gas        std          two   
2                  NaN  alfa-romero       gas        std          two   
3                164.0         audi       gas        std         four   
4                164.0         audi       gas        std         four   
..                 ...          ...       ...        ...          ...   
200               95.0        volvo       gas        std         four   
201               95.0        volvo       gas      turbo         four   
202               95.0        volvo       gas        std         four   
203               95.0        volvo    diesel      turbo         four   
204               95.0        volvo       gas      turbo         four   

      body-style drive-wheels engine-location  wheel-base  length  ...  \
0    convertible          rwd           front        88.6   168.8  ...   
1    convertible          rwd           front        88.6   168.8  ...   
2      hatchback          rwd           front        94.5   171.2  ...   
3          sedan          fwd           front        99.8   176.6  ...   
4          sedan          4wd           front        99.4   176.6  ...   
..           ...          ...             ...         ...     ...  ...   
200        sedan          rwd           front       109.1   188.8  ...   
201        sedan          rwd           front       109.1   188.8  ...   
202        sedan          rwd           front       109.1   188.8  ...   
203        sedan          rwd           front       109.1   188.8  ...   
204        sedan          rwd           front       109.1   188.8  ...   

     fuel-system  bore  stroke compression-ratio horsepower  peak-rpm  \
0           mpfi  3.47    2.68               9.0      111.0    5000.0   
1           mpfi  3.47    2.68               9.0      111.0    5000.0   
2           mpfi  2.68    3.47               9.0      154.0    5000.0   
3           mpfi  3.19    3.40              10.0      102.0    5500.0   
4           mpfi  3.19    3.40               8.0      115.0    5500.0   
..           ...   ...     ...               ...        ...       ...   
200         mpfi  3.78    3.15               9.5      114.0    5400.0   
201         mpfi  3.78    3.15               8.7      160.0    5300.0   
202         mpfi  3.58    2.87               8.8      134.0    5500.0   
203          idi  3.01    3.40              23.0      106.0    4800.0   
204         mpfi  3.78    3.15               9.5      114.0    5400.0   

    city-mpg  highway-mpg    price  symboling  
0         21           27  13495.0          3  
1         21           27  16500.0          3  
2         19           26  16500.0          1  
3         24           30  13950.0          2  
4         18           22  17450.0          2  
..       ...          ...      ...        ...  
200       23           28  16845.0         -1  
201       19           25  19045.0         -1  
202       18           23  21485.0         -1  
203       26           27  22470.0         -1  
204       19           25  22625.0         -1  

[205 rows x 26 columns]